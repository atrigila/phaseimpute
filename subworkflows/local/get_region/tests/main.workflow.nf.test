nextflow_workflow {

    name "Test Workflow GET_REGION"
    script "../main.nf"
    workflow "GET_REGION"
    tag 'subworkflows'
    tag 'get_region'
    tag 'subworkflows/get_region'

    test("Should run with 'all'") {

        when {
            workflow {
                """
                input[0] = "all"
                input[1] = Channel.of([
                    [genome:"GRCh37"],
                    file("https://raw.githubusercontent.com/LouisLeNezet/test-datasets/phaseimpute/data/reference_genome/21_22/hs38DH.chr21_22.s.fa", checkIfExists: true),
                    file("https://raw.githubusercontent.com/LouisLeNezet/test-datasets/phaseimpute/data/reference_genome/21_22/hs38DH.chr21_22.s.fa.fai", checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.regions).match() }
            )
        }
    }

    test("Should run with specified chr") {

        when {
            workflow {
                """
                input[0] = "chr22"
                input[1] = Channel.of([
                    [genome:"GRCh37"],
                    file("https://raw.githubusercontent.com/LouisLeNezet/test-datasets/phaseimpute/data/reference_genome/21_22/hs38DH.chr21_22.s.fa", checkIfExists: true),
                    file("https://raw.githubusercontent.com/LouisLeNezet/test-datasets/phaseimpute/data/reference_genome/21_22/hs38DH.chr21_22.s.fa.fai", checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.regions).match() }
            )
        }
    }

    test("Should run with specified region without fasta") {

        when {
            workflow {
                """
                input[0] = "chr22:0-4000"
                input[1] = Channel.of([[],[],[]])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.regions).match() }
            )
        }
    }
}
