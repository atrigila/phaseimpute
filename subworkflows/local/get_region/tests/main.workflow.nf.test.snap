{
    "Should run with specified region without fasta": {
        "content": [
            [
                [
                    {
                        "chr": "chr22",
                        "region": "chr22:0-4000"
                    },
                    "chr22:0-4000"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-17T15:20:16.458964814"
    },
    "Should run with specified chr": {
        "content": [
            [
                [
                    {
                        "chr": "chr22",
                        "region": "chr22:16570000-16610000"
                    },
                    "chr22:16570000-16610000"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-17T15:20:11.51328291"
    },
    "Should run with 'all'": {
        "content": [
            [
                [
                    {
                        "chr": "chr21",
                        "region": "chr21:16570000-16610000"
                    },
                    "chr21:16570000-16610000"
                ],
                [
                    {
                        "chr": "chr22",
                        "region": "chr22:16570000-16610000"
                    },
                    "chr22:16570000-16610000"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-17T15:20:06.490072121"
    }
}